### PR TITLE
KAFKA-16550: add integration test for LogDirsCommand

### PR DESCRIPTION
**Pull request by FrankYang0529**
Add integration test cases for LogDirsCommand:
* testLogDirsWithoutBrokers 
* testLogDirsWithBrokers
* testLogDirsWithNonExistentTopic
* testLogDirsWithSpecificTopic

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: tools/src/test/java/org/apache/kafka/tools/LogDirsCommandTest.java [hunk 1]
```diff
@@ -16,33 +16,155 @@
...
+                });
+            });
+        }
+    }
+
+    @ClusterTest
+    public void testLogDirsWithNonExistentTopic(ClusterInstance clusterInstance) {
+        try (Admin admin = Admin.create(Collections.singletonMap(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, clusterInstance.bootstrapServers()))) {
+            String output = assertDoesNotThrow(() -> execute(fromArgsToOptions("--bootstrap-server", clusterInstance.bootstrapServers(), "--topic-list", TOPIC, "--describe"), admin));
+            System.out.println("output: " + output);
```
Comment (m1a2st) [inline tools/src/test/java/org/apache/kafka/tools/LogDirsCommandTest.java]: Please remove debug message.

File: tools/src/test/java/org/apache/kafka/tools/LogDirsCommandTest.java [hunk 2]
```diff
@@ -16,33 +16,155 @@
...
+            });
+        }
+    }
+
+    @ClusterTest
+    public void testLogDirsWithSpecificTopic(ClusterInstance clusterInstance) {
+        createTopic(clusterInstance, TOPIC);
+        try (Admin admin = Admin.create(Collections.singletonMap(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, clusterInstance.bootstrapServers()))) {
+            String output = assertDoesNotThrow(() -> execute(fromArgsToOptions("--bootstrap-server", clusterInstance.bootstrapServers(), "--topic-list", TOPIC, "--describe"), admin));
+            System.out.println("output: " + output);
```
Comment (m1a2st) [inline tools/src/test/java/org/apache/kafka/tools/LogDirsCommandTest.java]: Please remove debug message.

Comment (FrankYang0529) [reply tools/src/test/java/org/apache/kafka/tools/LogDirsCommandTest.java]: Remove both message. Thanks.

File: tools/src/test/java/org/apache/kafka/tools/LogDirsCommandTest.java [hunk 3]
```diff
@@ -16,33 +16,153 @@
...
+                    logDirInfoValue.replicaInfos().forEach((topicPartition, replicaInfo) -> {
+                        assertFalse(output.contains("\"partition\":\"" + topicPartition + "\""));
+                    });
+                });
+            });
+        }
+    }
+
+    @ClusterTest
+    public void testLogDirsWithSpecificTopic(ClusterInstance clusterInstance) {
```
Comment (chia7712) [inline tools/src/test/java/org/apache/kafka/tools/LogDirsCommandTest.java:124]: Could you please add test to verify that it returns only the specific topic will get returned? for example, you can create another topic before testing.

File: tools/src/test/java/org/apache/kafka/tools/LogDirsCommandTest.java [hunk 4]
```diff
@@ -16,33 +16,153 @@
...
+
+            // check all brokers are present
+            clusterInstance.brokerIds().forEach(brokerId -> {
+                assertTrue(output.contains("\"broker\":" + brokerId));
+            });
+
+            // check all log dirs and topic partitions are present
+            Map<Integer, Map<String, LogDirDescription>> logDirs = assertDoesNotThrow(() -> admin.describeLogDirs(clusterInstance.brokerIds()).allDescriptions().get());
+            assertFalse(logDirs.isEmpty());
+            logDirs.forEach((brokerId, logDirInfo) -> {
```
Comment (chia7712) [inline tools/src/test/java/org/apache/kafka/tools/LogDirsCommandTest.java]: please remove `{}`

File: tools/src/test/java/org/apache/kafka/tools/LogDirsCommandTest.java [hunk 5]
```diff
@@ -16,33 +16,153 @@
...
+            // check only broker in list is present
+            assertTrue(output.contains("\"broker\":" + brokerId));
+            clusterInstance.brokerIds().stream().filter(id -> id != brokerId).forEach(id -> {
+                assertFalse(output.contains("\"broker\":" + id));
+            });
+
+            // check log dir and topic partition are present
+            Map<Integer, Map<String, LogDirDescription>> logDirs = assertDoesNotThrow(() -> admin.describeLogDirs(Collections.singleton(brokerId)).allDescriptions().get());
+            assertEquals(1, logDirs.size());
+            logDirs.forEach((brokerIdValue, logDirInfo) -> {
```
Comment (chia7712) [inline tools/src/test/java/org/apache/kafka/tools/LogDirsCommandTest.java:91]: please remove `{}`

File: tools/src/test/java/org/apache/kafka/tools/LogDirsCommandTest.java [hunk 6]
```diff
@@ -16,33 +16,153 @@
...
+            String output = assertDoesNotThrow(() -> execute(fromArgsToOptions("--bootstrap-server", clusterInstance.bootstrapServers(), "--topic-list", TOPIC, "--describe"), admin));
+            // check all brokers are present
+            clusterInstance.brokerIds().forEach(brokerId -> {
+                assertTrue(output.contains("\"broker\":" + brokerId));
+            });
+
+            // check log dir is present, but topic partition is not
+            Map<Integer, Map<String, LogDirDescription>> logDirs = assertDoesNotThrow(() -> admin.describeLogDirs(clusterInstance.brokerIds()).allDescriptions().get());
+            assertFalse(logDirs.isEmpty());
+            logDirs.forEach((brokerId, logDirInfo) -> {
```
Comment (chia7712) [inline tools/src/test/java/org/apache/kafka/tools/LogDirsCommandTest.java]: please remove `{}`

**Review by chia7712 — COMMENTED** [review]
@FrankYang0529 thanks for this patch

**Comment by FrankYang0529** [comment]
Hi @chia7712, I address all comments. Could you take a look again? Thank you.

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request merged**